workflow/metrics/metrics.go

```diff
@@ -147,3 +153,40 @@ func (m *Metrics) OperationPanic() {
 func (m *Metrics) CronWorkflowSubmissionError() {
 	m.errors[ErrorCauseCronWorkflowSubmissionError].Inc()
 }
+
+// Act as a metrics provider for a workflow queue
+var _ workqueue.MetricsProvider = &Metrics{}
+
+func (m *Metrics) NewDepthMetric(name string) workqueue.GaugeMetric {
+	key := fmt.Sprintf("%s-depth", name)
+	if _, ok := m.workqueueMetrics[key]; !ok {
+		m.workqueueMetrics[key] = newGauge("queue_depth_count", "Depth of the queue", map[string]string{"queue_name": name})
+	}
+	return m.workqueueMetrics[key].(prometheus.Gauge)
+}
+
+func (m *Metrics) NewAddsMetric(name string) workqueue.CounterMetric {
+	key := fmt.Sprintf("%s-adds", name)
+	if _, ok := m.workqueueMetrics[key]; !ok {
+		m.workqueueMetrics[key] = newCounter("queue_adds_count", "Adds to the queue", map[string]string{"queue_name": name})
+	}
+	return m.workqueueMetrics[key].(prometheus.Counter)
+}
+
+// These metrics are not relevant to be exposed
+type noopMetric struct{}
+
+func (noopMetric) Inc()            {}
+func (noopMetric) Dec()            {}
+func (noopMetric) Set(float64)     {}
+func (noopMetric) Observe(float64) {}
+
+func (m *Metrics) NewLatencyMetric(name string) workqueue.HistogramMetric      { return noopMetric{} }
+func (m *Metrics) NewRetriesMetric(name string) workqueue.CounterMetric        { return noopMetric{} }
+func (m *Metrics) NewWorkDurationMetric(name string) workqueue.HistogramMetric { return noopMetric{} }
```

Comment:
Just noting that we actually do track this now (somewhat) with the `operation_duration_seconds` metric.  Wondering if it might be worth considering refactoring the code to use this and eliminate any custom code since in addition to the workflow operator duration metrics, you would get pod operation duration metrics for free.

Reply:
Yes, that was my reason for not including it. `operation_duration_seconds` is more meaningful. Pod opeartion metrics are actually insignificant, since all the work done there is extract the Workflow name and add it to the workflow queue.